systemFlake.nix

```diff
@@ -161,7 +160,7 @@ otherArguments
     optional = check: optionalAttrs (check != null);
   in
   { inherit pkgs; }
-  // optional packagesBuilder { packages = packagesBuilder pkgs; }
+  // optional packagesBuilder { packages = packagesBuilder pkgs channels; }
```

Comment:
Why does channels need to be passed here? Is there any way this can be avoided, perhaps by passing channels as the first argument to overlaysPackagesBuilder, and users can do the same - pass their channels to create the packagesBuilder.
I don't like having to change the api for this builder.

Reply:
I also don't like it. I'm going to think a little harder. Thanks!